core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -424,6 +424,7 @@ public Location add(Operation operation) throws IOException {
             closeOnTragicEvent(ex);
             throw ex;
         } catch (Throwable e) {
```

Comment:
I'm a bit hesitant to be so strict when the serialization to an in memory buffer is under the try. I think we should split it in two - any exception up to when we write to the file is just an exception, the rest is tragic. Makes sense?